secure/storage/src/vault.rs

```diff
@@ -175,4 +215,80 @@ impl KVStorage for VaultStorage {
     }
 }
 
-impl CryptoKVStorage for VaultStorage {}
+impl CryptoStorage for VaultStorage {
+    fn create_key(&mut self, name: &str, policy: &Policy) -> Result<Ed25519PublicKey, Error> {
+        let ns_name = self.crypto_name(name);
+        match self.get_public_key(name) {
+            Ok(_) => return Err(Error::KeyAlreadyExists(ns_name)),
+            Err(Error::KeyNotSet(_)) => (/* Expected this for new keys! */),
+            Err(e) => return Err(e),
+        }
+
+        self.client.create_ed25519_key(&ns_name, true)?;
+        self.set_policies(&ns_name, policy)?;
+        self.get_public_key(name).map(|v| v.public_key)
+    }
+
+    fn export_private_key(&self, name: &str) -> Result<Ed25519PrivateKey, Error> {
+        let name = self.crypto_name(name);
+        Ok(self.client.export_ed25519_key(&name, None)?)
+    }
+
+    fn export_private_key_for_version(
+        &self,
+        name: &str,
+        version: Ed25519PublicKey,
+    ) -> Result<Ed25519PrivateKey, Error> {
+        let name = self.crypto_name(name);
+        let vers = self.key_version(&name, &version)?;
+        Ok(self.client.export_ed25519_key(&name, Some(vers))?)
+    }
+
+    fn get_public_key(&self, name: &str) -> Result<PublicKeyResponse, Error> {
+        let name = self.crypto_name(name);
+        let resp = self.client.read_ed25519_key(&name)?;
+        let mut last_key = resp.first();
+        for key in &resp {
+            last_key = if let Some(ckey) = last_key {
+                if ckey.version > key.version {
+                    Some(&ckey)
+                } else {
+                    Some(&key)
+                }
+            } else {
+                Some(&key)
```

Comment:
Unless I'm missing something, I don't see how we could ever get to this case? i.e., if resp.first() is None, then we shouldn't enter into the loop? Can we not just remove the 'else' branch here?